### ISSUE TITLE
Issue discovered on WireMock's `MemoizingMatchResult` with `Use lambdas where possible`

### ISSUE THREAD

**Issue by timtebeek** [moderneinc/support-public#31]
### Problem
> Describe the issue you are experiencing.

### Expected behavior
> Describe what you expected to see.

### Example diff

**From:** `src/main/java/com/github/tomakehurst/wiremock/matching/MemoizingMatchResult.java`

```diff
 
private final Supplier<Double> memoizedDistance =
 Suppliers.memoize(
-          new Supplier<Double>() {
-            @Override
-            public Double get() {
-              return target.getDistance();
-            }
-          });
+          () -> target.getDistance());

private final Supplier<Boolean> memoizedExactMatch =
 Suppliers.memoize(
-          new Supplier<Boolean>() {
-            @Override
-            public Boolean get() {
-              return target.isExactMatch();
-            }
-          });
+          () -> target.isExactMatch());

private final MatchResult target;
```

#### Recipes in example diff:
- `org.openrewrite.java.cleanup.UseLambdaForFunctionalInterface`

### References:
- [View original result](https://public.moderne.io/results/0xoDM/details/eyJfX3R5cGVuYW1lIjoiR2l0SHViUmVwb3NpdG9yeSIsIm9yaWdpbiI6ImdpdGh1Yi5jb20iLCJwYXRoIjoid2lyZW1vY2svd2lyZW1vY2siLCJicmFuY2giOiJtYXN0ZXIifQ==?referrer=%2Fresults%2F0xoDM)
- Recipe ID: `org.openrewrite.java.cleanup.UseLambdaForFunctionalInterface`
- Recipe Name: `Use lambdas where possible`
- Repository: `wiremock/wiremock/master`
- Created at Fri Mar 24 2023 15:33:42 GMT+0100 (Central European Standard Time)

**Comment by timtebeek** [comment]
Fails with:
```
> Task :compileJava
/home/tim/Documents/workspace/wiremock/src/main/java/com/github/tomakehurst/wiremock/matching/MemoizingMatchResult.java:25: error: illegal forward reference
          () -> target.getDistance());
                ^
/home/tim/Documents/workspace/wiremock/src/main/java/com/github/tomakehurst/wiremock/matching/MemoizingMatchResult.java:29: error: illegal forward reference
          () -> target.isExactMatch());
                ^
```
And in IDEA with `Variable 'target' might not have been initialized`.

**Comment by knutwannheden** [comment]
Interesting. How is it now that we move issues to the openrewrite organization?

**Comment by jbessels** [comment]
**When will this be moved to  1.19.0-SNAPSHOT?** Tim basically handpicked this bug for me - represents something which is done in my code. Just to be sure Maven would pick it I removed the existing 1.19.0-SNAPSHOT on my system. 

When running the recipe I'm getting the exact some error as before.

**Comment by timtebeek** [comment]
The fix was in https://github.com/openrewrite/rewrite/commit/56c1466c61e20354f493fccec74082bb47f00237 ; so I'd expect it to be available in rewrite-java `7.39.0-SNAPSHOT`.
